File: javascript/xss.js
const express = require('express');
const app = express();

const escapeHtml = (str) => {
  return str.replace(/[&<>"']/g, (m) => {
    return {'&': '&amp;', '<': '&lt;', '>': '&gt;', '"': '&quot;', "'": '&#39;'}[m];
  });
};

app.get('/greet', (req, res) => {
  const name = req.query.name || '';
  // CWE-79: user input reflected in HTML without escaping.
  res.send(`<h1>Hello, ${escapeHtml(name)}!</h1>`);
});

app.listen(3000);

File: python/xss.py
from flask import Flask, request
from markupsafe import escape

app = Flask(__name__)

@app.route('/greet')
def greet():
    name = request.args.get('name', '')
    # CWE-79: user input reflected in HTML response without escaping.
    return f'<h1>Hello, {escape(name)}!</h1>'

if __name__ == '__main__':
    app.run()
